examples/controls_example/public/react_controls/data_controls/search_control/get_search_control_factory.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect } from 'react';
+import deepEqual from 'react-fast-compare';
+import { BehaviorSubject, combineLatest, debounceTime, distinctUntilChanged } from 'rxjs';
+
+import { EuiFieldSearch, EuiFormRow, EuiRadioGroup } from '@elastic/eui';
+import { CoreStart } from '@kbn/core/public';
+import { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { useStateFromPublishingSubject } from '@kbn/presentation-publishing';
+
+import { initializeDataControl } from '../initialize_data_control';
+import { DataControlFactory } from '../types';
+import {
+  SearchControlApi,
+  SearchControlState,
+  SearchControlTechniques,
+  SEARCH_CONTROL_TYPE,
+} from './types';
+
+const allSearchOptions = [
+  {
+    id: 'match',
+    label: i18n.translate('controlsExamples.searchControl.searchTechnique.match', {
+      defaultMessage: 'Fuzzy match',
+    }),
+    'data-test-subj': 'searchControl__matchSearchOptionAdditionalSetting',
+  },
+  {
+    id: 'simple_query_string',
+    label: i18n.translate('controlsExamples.searchControl.searchTechnique.simpleQueryString', {
+      defaultMessage: 'Query string',
+    }),
+    'data-test-subj': 'optionsListControl__queryStringSearchOptionAdditionalSetting',
+  },
+];
+
+const DEFAULT_SEARCH_TECHNIQUE = 'match';
+
+export const getSearchEmbeddableFactory = ({
+  core,
+  dataViewsService,
+}: {
+  core: CoreStart;
+  dataViewsService: DataViewsPublicPluginStart;
+}): DataControlFactory<SearchControlState, SearchControlApi> => {
+  return {
+    type: SEARCH_CONTROL_TYPE,
+    getIconType: () => 'search',
+    getDisplayName: () =>
+      i18n.translate('controlsExamples.searchControl.displayName', { defaultMessage: 'Search' }),
+    isFieldCompatible: (field) => {
+      return (
+        field.searchable &&
+        field.spec.type === 'string' &&
+        (field.spec.esTypes ?? []).includes('text')
+      );
+    },
+    CustomOptionsComponent: ({ stateManager }) => {
```

Comment:
Looks like we're tightly coupling the options component to this type of state manager. Should we make this more explicit by building the state manager automatically?

Reply:
I'm a bit confused on what you mean - do you mean making the type of `stateManager` something like this

```typescript
stateManager: ControlStateManager<Omit<State, keyof DefaultDataControlState>>;
```

and then manually removing all `DefaultDataControlState` from the state manager when we build `CustomSettingsComponent`?

Reply:
This can be done in a followup, but the question was more long-term. The fact that the `stateManager` object is taken in here means we're tightly coupling Controls to an RXJS based state manager. This is completely fine, and even expected, but it means that we might be able to do make some extra sugar here by generating the state manager automatically. No action items for the time being though!